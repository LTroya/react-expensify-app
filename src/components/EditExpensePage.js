import React, { Component } from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

export class EditExpensePage extends Component {
    handleEditExpense = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };

    handleRemoveExpense = () => {
        this.props.removeExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <h1>Edit expense</h1>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.handleEditExpense}
                />

                <button onClick={this.handleRemoveExpense}>
                    Remove
                </button>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id),
});

const mapDispatchToProps = (dispatch) => ({
    editExpense: (id, expense) => {
        dispatch(editExpense(id, expense));
    },
    removeExpense: (id) => {
        dispatch(removeExpense({ id }));
    },
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
