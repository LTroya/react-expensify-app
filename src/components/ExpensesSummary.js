import React from 'react';
import {Link} from 'react-router-dom';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getVisiblesExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal).format('$0,0.00');

    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expensesCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = getVisiblesExpenses(state.expenses, state.filters);
    const total = getExpensesTotal(visibleExpenses) / 100;

    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: total,
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
