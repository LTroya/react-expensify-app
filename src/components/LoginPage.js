import React from 'react';
import { connect } from 'react-redux';
import {
    startLoginWithFacebook,
    startLoginWithGithub,
    startLoginWithGoogle,
    startLoginWithTwitter,
} from '../actions/auth';

export const LoginPage = ({ startLoginWithGoogle, startLoginWithFacebook, startLoginWithTwitter, startLoginWithGithub }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control.</p>
            <button className="button button--google" onClick={startLoginWithGoogle}>Login with Google</button>
            <button className="button button--facebook" onClick={startLoginWithFacebook}>Login with Facebook</button>
            <button className="button button--twitter" onClick={startLoginWithTwitter}>Login with Twitter</button>
            <button className="button button--github" onClick={startLoginWithGithub}>Login with Github</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLoginWithGoogle: () => dispatch(startLoginWithGoogle()),
    startLoginWithFacebook: () => dispatch(startLoginWithFacebook()),
    startLoginWithTwitter: () => dispatch(startLoginWithTwitter()),
    startLoginWithGithub: () => dispatch(startLoginWithGithub()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
