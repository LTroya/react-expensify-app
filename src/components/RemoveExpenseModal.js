import React from 'react';
import Modal from 'react-modal';

const RemoveExpensesModal = ({handleRemoveRequested, handleHideModal, handleRemoveExpense, showModal, expense}) => (
    <Modal
        isOpen={showModal}
        onRequestClose={handleHideModal}
        contentLabel="Remove expense"
        closeTimeoutMS={200}
        className="modal">
        <h3 className="modal__title">Remove expense</h3>
        Are you sure you want to remove <strong>{expense.description}</strong> expense?
        <div className="modal__actions">
            <button className="button" onClick={handleRemoveExpense}>Remove</button>
            <button className="button button--secondary" onClick={handleHideModal}>Cancel</button>
        </div>
    </Modal>
);

export default RemoveExpensesModal;
