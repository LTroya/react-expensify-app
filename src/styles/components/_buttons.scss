.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  margin-bottom: $m-size;
  padding: $s-size;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888;
}

.button--google { background: #dd4b39; }
.button--facebook { background: #3b5999; }
.button--twitter { background: #55acee; }
.button--github { background: #34465d; }
