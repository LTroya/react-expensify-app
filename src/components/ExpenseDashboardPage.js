import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpensesListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary/>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;
