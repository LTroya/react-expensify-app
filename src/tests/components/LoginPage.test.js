import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { LoginPage } from '../../components/LoginPage';

let wrapper, startLogin;

beforeEach(() => {
    startLogin = jest.fn();
    wrapper = shallow(<LoginPage startLogin={startLogin}/>);
})

test('Should render Header correctly', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
});

test('Should startLogin on button click', () => {
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});
