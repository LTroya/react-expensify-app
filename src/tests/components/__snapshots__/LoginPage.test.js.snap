// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      It's time to get your expenses under control.
    </p>
    <button
      className="button button--google"
      onClick={[Function]}
    >
      Login with Google
    </button>
    <button
      className="button button--facebook"
      onClick={[Function]}
    >
      Login with Facebook
    </button>
    <button
      className="button button--twitter"
      onClick={[Function]}
    >
      Login with Twitter
    </button>
    <button
      className="button button--github"
      onClick={[Function]}
    >
      Login with Github
    </button>
  </div>
</div>
`;
